tests/unit/test_settings_logs.py

```
@@ -41,3 +41,85 @@ def test_api_logs_rate_limit(client, monkeypatch):
+    data = resp.get_json()
+    assert data["truncated"] is True
+    assert data["meta"]["limit"] == 2000
+
+
+def test_api_logs_response_size_guardrail(client, monkeypatch):
+    """Huge output is truncated under MAX_RESPONSE_BYTES."""
+    import blueprints.settings as mod
+
+    # Generate huge log output
+    big_lines = ["X" * 1000 for _ in range(2000)]
+    monkeypatch.setattr(mod, "_read_log_lines", lambda h: big_lines)
```

Ruff (0.15.7)

[warning] Unused lambda argument: `h`

(ARG005)